lib/Conversion/TorchToLinalg/Linear.cpp

```diff
@@ -302,10 +302,59 @@ class ConvertAtenMatmulOp : public OpConversionPattern<AtenMatmulOp> {
     auto lhsType = lhs.getType().cast<RankedTensorType>();
     auto rhsType = rhs.getType().cast<RankedTensorType>();
 
+    auto lhsTorchType = op.getSelf().getType().cast<ValueTensorType>();
+    auto rhsTorchType = op.getOther().getType().cast<ValueTensorType>();
+
     // Get the rank of both matrix.
     unsigned lhsRank = lhsType.getRank();
     unsigned rhsRank = rhsType.getRank();
 
+    Value lhsZeroPoint, rhsZeroPoint;
+    getZeroPoint(op.getSelf(), lhsZeroPoint);
+    getZeroPoint(op.getOther(), rhsZeroPoint);
+
+    if (static_cast<bool>(lhsZeroPoint) != static_cast<bool>(rhsZeroPoint)) {
+      return rewriter.notifyMatchFailure(
+          op, "unsupported: aten.matmul with mixed quantization");
+    }
+
+    bool isUnsigned = torch_to_linalg::isUnsignedTorchType(lhsTorchType);
+    bool isUnsignedR = torch_to_linalg::isUnsignedTorchType(rhsTorchType);
+
+    if (!lhsZeroPoint) {
```

Comment:
Split this nested if into an unnested versions. Namely

if (!lhsZeroPoint && lhsTorchType.getDtype() != rhsTorchType.getDtype())
        return rewriter.notifyMatchFailure(
            op, "unsupported: aten.matmul with different input element types");
if (lhsZeroPoint) {
  ....
}